### ISSUE TITLE
pid of aplication does not always correspond to app number in apache log

### ISSUE THREAD

**Issue by naox** [phusion/passenger#1608]
pid of aplication does not always correspond to app number in apache log. As far as I can see there is no way to actualy read this mysterious aplication number to correlate lines of  apache log with actual aplication/aplication pid read from passenger-status. 

passenger-status

```
/home/hopsa/rails/redmine_test#default:
  App root: /home/hopsa/rails/redmine_test
  Requests in queue: 0
  * PID: 16249   Sessions: 0       Processed: 1       Uptime: 6s
    CPU: 0%      Memory  : 32M     Last used: 6s ago
```

apache log:

```
App 11733 stderr: [ 2015-09-15 10:57:25.1729 16249/0x007f493316bdf8(Worker 1) utils.rb:84 ]: *** Exception RuntimeError in Rack application object (Missing `secret_token` and `secret_key_base` for 'production' environment, set these values in `config/secrets.yml`) (process 16249, thread 0x007f493316bdf8(Worker 1)):
App 11733 stderr:       from /home/hopsa/.gems/2.2/gems/railties-4.2.3/lib/rails/application.rb:534:in `validate_secret_key_config!'
```

**Comment by OnixGH** [comment]
The app number is not supposed to be the same as the PID. If you look to the right in the first logline you can see the PID 16249, that's how you can correlate.

**Comment by naox** [comment]
Every log line should be correlatable on its own because there is no separators for log entries from diffrent aplications and thus its impossible to correlate with 100% accuracy or correlate with automated tools. 
There is some aplication number in front of the log line. How to read it for all running aplications?

**Comment by OnixGH** [comment]
Woops, sorry, the close was by accident. What I was trying to say is that passenger loglines always contain the PID (after the date/time part), so for those lines correlation should not be an issue.

The App XXXX lines are (forwarded) output from the application itself, but I see what you're saying about the need to be able to correlate those too. I'll have a closer look.

**Comment by naox** [comment]
Most of the time App XXXX matches pid (for wsgi aplications at least). This however was ruby app (redmine) and it does not match.

If the App XXXX is forwarded from the aplication itself then maybe it would be good thing if passenger added own header to those log lines, header that could be customized and could for example contain: timestamp,aplication directory, aplication pid.

**Comment by OnixGH** [comment]
(also suggested in https://github.com/phusion/passenger/issues/1679)

**Comment by OnixGH** [comment]
The PID mismatch is probably caused by the use of smart spawning (i.e. the PID will be the preloader PID). 

Workaround: if it's super important to have the match, then you could try using direct spawning instead. There is a small performance hit, mostly when using autoscaling (spawning apps is a bit slower).

Regarding timestamp and app info: wrapping the application output with a passenger core timestamp is not a good idea because the timestamp will be that of when the log was received by the passenger core, not the time when the logline was generated (i.e. can be misleading). The application is already free to do its own logging anyway so it's better equipped to do this.

**Comment by FooBarWidget** [comment]
The original issue (mismatching PID) has been fixed as of Passenger 5.3.
